In [128]:
library(tidyverse)
library(dplyr)

## 1)  Data Description:

In [125]:
players<- read_csv("players (2).csv")

Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


In [121]:
nrow(players)
ncol(players)
names(players)

experience_players<- players |>
  group_by(experience) |>
  summarise(
    mean_played_hours = mean(played_hours, na.rm = TRUE),
    mean_age = mean(Age, na.rm = TRUE),
    number_of_obs = n())
experience_players

summary_age <- players |>
  summarise(age_mean = mean(Age, na.rm = TRUE),
  played_hours_mean = mean(played_hours, na.rm = TRUE),
            age_min = min(Age, na.rm=TRUE),
            age_max = max(Age, na.rm = TRUE))
summary_age

summary_played <- players |>
  summarise(played_hours_mean = mean(played_hours, na.rm = TRUE),
  played_hours_mean = mean(played_hours, na.rm = TRUE),
            played_hours_min = min(played_hours, na.rm=TRUE),
            played_hours_max = max(played_hours, na.rm = TRUE))
summary_played

sub_players<- players |>
  group_by(subscribe) |>
  summarise(
    mean_played_hours = mean(played_hours, na.rm = TRUE),
    mean_age = mean(Age, na.rm = TRUE),
    number_of_obs = n())
sub_players

gender_players<- players |>
  group_by(gender) |>
  summarise(
    mean_played_hours = mean(played_hours, na.rm = TRUE),
    mean_age = mean(Age, na.rm = TRUE),
    number_of_obs = n())
gender_players

summary <- players |>
  summarise(age_mean = mean(Age, na.rm = TRUE),
  played_hours_mean = mean(played_hours, na.rm = TRUE),
            age_min = min(Age, na.rm=TRUE),
            age_max = max(Age, na.rm = TRUE),
played_hours_mean = mean(played_hours, na.rm = TRUE),
  played_hours_mean = mean(played_hours, na.rm = TRUE),
            played_hours_min = min(played_hours, na.rm=TRUE),
            played_hours_max = max(played_hours, na.rm = TRUE))
summary

[1] 196

[1] 7

[1] "experience"   "subscribe"    "hashedEmail"  "played_hours" "name"        
[6] "gender"       "Age"

experience mean_played_hours mean_age number_of_obs
1 Amateur     6.0174603        21.36508 63           
2 Beginner    1.2485714        21.65714 35           
3 Pro         2.6000000        16.92308 14           
4 Regular    18.2083333        22.02857 36           
5 Veteran     0.6479167        20.95833 48

age_mean played_hours_mean age_min age_max
1 21.13918 5.845918          9       58

played_hours_mean played_hours_min played_hours_max
1 5.845918          0                223.1

subscribe mean_played_hours mean_age number_of_obs
1 FALSE     0.5019231         23.7500   52          
2  TRUE     7.7756944         20.1831  144

gender            mean_played_hours mean_age number_of_obs
1 Agender            6.25000000       23.50000   2          
2 Female            10.63513514       21.81081  37          
3 Male               4.12741935       20.85484 124          
4 Non-binary        14.88000000       19.06667  15          
5 Other              0.20000000            NaN   1          
6 Prefer not to say  0.37272727       25.00000  11          
7 Two-Spirited       0.08333333       20.00000   6

age_mean played_hours_mean age_min age_max played_hours_min played_hours_max
1 21.13918 5.845918          9       58      0                223.1

### Players Data Set

- **Observations**: 196
- **Variables**: 7 
    - 'experience','subscribe','hashedEmail','played_hours','name','gender','Age'
        - **experience**: 5 types of experiences
            - 'Amateur', ' Beginner', 'Pro', 'Regular', 'Veteran'
        - **Age**: average is 21, max is 58, min is 
        - **Played hours**: average is 5.85 hours, max is 223.1 hours
        - **hashedEmail**: is a the unique way to adress each person
        - **Subscribed**: people who were subscribed to gameing played and average of 7.27 more hours
        - **Gender**: there are 7 catagorys of gender
            - 'Agender', 'Female', 'Male', 'Non-bonary', 'Other', 'Two-Spirited','Prefer not to say'
            - majority were male, non-binary people played the most, highest mean age was of those who did not specify gender
- **Issues**:
    - There are many people who did not play at all and still have a profile; this may affect the distribution and data analaysis. However, it can solved by filtering those individuals out.
    - There are many columns with NA as a value, this will definately distube our imformtion but can be solved using na.rm=TRUE


In [122]:
sessions<- read_csv("sessions (2).csv")

tidy_sessions<- sessions|>
separate(start_time, c("start_date", "start_time"), " ")|>
separate(end_time, c("end_date", "end_time"), " ")

Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


In [123]:
nrow(sessions)
ncol(sessions)
names(sessions)

time<- tidy_sessions|>
summarise(start_time_mean = mean(original_start_time, na.rm = TRUE),
  end_time_mean = mean(original_end_time, na.rm = TRUE))
time


[1] 1535

[1] 5

[1] "hashedEmail"         "start_time"          "end_time"           
[4] "original_start_time" "original_end_time"

start_time_mean end_time_mean
1 1.719201e+12    1.719196e+12

### Sessions Data Set

- **Observations**: 1535
- **Variables**: 5
    - 'hashedEmail', 'start_time', 'end_time', 'original_start_time', 'original_end_time'
    - Most of these variables are characters so you can't find numerical summaries of them easily
    - Orignal start time and end time are in UNIX time, so to convert it into readable form you much do some steps.
- **Issues**:
    - The start_time and end_time are not in tidy form because one variable has both data and time, so I used separate() to split the columns into two so we have in total 7 columns now.
    - Most of the columns are in character format so it is difficult to summarize the data.
        - I can only use original end and start time, which are in milliseconds. 

## 2) Questions:

**Broad Question**: What player characteristics and behaviours are most predictive of subscribing to a game-related newsletter, and how do these features differ between various player types?

**Specific Question**: Can age, and hours played predict whether they subscribe to game related newsletter in players.csv, and will this change depending on experience level?

**How the data will help answer this question**: In the players.csv data, there are variables related to both player demographic and game usage. Using the variables 'Age' and 'played_hours', I can predict which kind of people are most likely to subscribe to game-related newsletters using classification. Furthermore, I can separate the data using experience levels to check if there are any trends within different levels of expertise. 

**How will I wrangle the data**:

1) *Load*:
    - load player.csv into my notebook
3) *Select and Clean*: select only the necessary columns such as Age, played_hours, experience, and subscribe
   - Ensure data is usable by filtering out NA values and invalid observations (ex. 0 hours played)
4) *Data Analysis and Visualizations*:
   -  Make a scatter plot of Age vs Played_hours, with the subscription variable coloured
        - This is to find a general trend of Age and Played_hours
   - Facet 'experience' to see if the pattern is different for each player type
        - This way I can compare the trends between different experiences and how they differ from the general trend as well.
6) *Classification*:
     - Check for imbalance in data (compare the amount of subscribed to not subscribed)
   - I can also use the general data set to make a classification model to check if age and played_hours true is able to detect if a person subscribes or not.


## 3)  Exploratory Data Analysis and Visualization

Demonstrate that the dataset can be loaded into R.

- Do the minimum necessary wrangling to turn your data into a **tidy format**. 
    - Do not do any additional wrangling here; that will happen later during the group project phase.
- Compute the **mean value** for each quantitative variable in the players.csv data set. Report the mean values in a table format.
Make a few exploratory visualizations of the data to help you understand it.
Use our visualization best practices to make high-quality plots (make sure to include labels, titles, units of measurement, etc)
Explain any insights you gain from these plots that are relevant to address your question